security/try-node-to-node-encryption.md

```diff
@@ -0,0 +1,67 @@
+description: Try out enabling WireGuard for state-of-the-art cryptographic security between pods for Calico clusters.
+---
+
+### Big picture
+
+Enable WireGuard to secure node-to-node traffic in a {{site.prodname}} cluster.
+
+> **Warning!** Node-to-node encryption is a tech preview and should not be used in production clusters. The functionality is not supported with overlay networks (in particular, NodePort) and using NodePort from within a non host-networked will not work. It has had very limited testing and it will contain bugs (please report these on the Calico Users slack or GitHub). In addition, it does not support all the features of {{site.prodname}}.
+{: .alert .alert-danger }
+
+### Value
+
+{{ site.prodname }} supports WireGuard tunnels between nodes providing transport-level security for node-to-node traffic. WireGuard provides [formally verified](https://www.wireguard.com/formal-verification/) secure and [performant tunnels](https://www.wireguard.com/performance/) without any specialized hardware. For a deep dive in to WireGuard implementation, see [whitepaper](https://www.wireguard.com/papers/wireguard.pdf).
+
+### Features
+
+This how-to guide uses the following {{site.prodname}} features:
+
+- **Felix configuration resource** with WireGuard configuration parameters.
+
+### Before you begin...
+
+- Verify the operating system(s) running on the nodes in the cluster [support WireGuard](https://www.wireguard.com/install/).
+
+    >**Note**: WireGuard in {{site.prodname}} does not support IPv6 at this time.
+    {: .alert .alert-info}
+- In the tech preview release, node-to-node encryption is supported on an underlying network that doesn’t require Calico to use an overlay. For example, a cluster with a routed network topology.
+
+### How to
+
+1. Install WireGuard on cluster nodes using these [instructions for your operating system](https://www.wireguard.com/install/).
+
+   >**Note**: Nodes that do not support WireGuard will not be secured by WireGuard tunnels, even if traffic running on the node to and from the pods goes to nodes that do support WireGuard.
+   {: .alert .alert-info}
+
+1. Enable WireGuard encryption across all the nodes using the following command.
+
+   ```
+   calicoctl patch felixconfiguration default --type='merge' -p '{"spec":{"wireguardEnabled":true}}'
+   ```
+
+For OpenShift, add the Felix configuration with WireGuard enabled [under custom resources]({{ site.baseurl }}/getting-started/openshift/installation#optionally-provide-additional-configuration).
+
+>**Note**: This above command can be used to change other WireGuard attributes. For a list of other WireGuard parameters and configuration evaluation, see the [Felix configuration]({{ site.baseurl }}/reference/resources/felixconfig#felix-configuration-definition).
+   {: .alert .alert-info}
+
+To assign an IPv4 address to WireGuard interface created by Felix, you need to update the [Node resource]({{ site.baseurl }}/reference/resources/node).
```

Comment:
I don't think we need to describe this since we aren't telling the user why they might want to do that, nor are we providing any example commands. It's likely that the assignment for this IP address will be automatic which will change the user requirements, so I think perhaps just leave this out for now.

WDYT?  Any reason you think we need to highlight this at the moment?

Reply:
Main reason for me is to tie this setting to the Encryption doc. If the admin would like to assign an IP address to WireGuard interface, this document guides where that setting can be found (especially since it's part of Node and not Felix configuration). I prefer to keep it here.

What do you mean by automatic address assignment for WireGuard interface? Please let me know if any information in reference to that will be useful.

Reply:
The assignment of the tunnel and vxlan interface addresses is automated. We would almost certainly want to automate the assignment of this address based on IP pool configuration.  This is not something we do yet and is possible the basis of the second phase of this feature.